src/trace/dma-trace.c

```diff
@@ -344,6 +356,14 @@ static int dma_trace_start(struct dma_trace_data *d)
 	if (err < 0)
 		return err;
 
+	/* Reset host buffer information as host is re-configuring dtrace */
+	key = k_spin_lock(&d->lock);
+
+	d->old_host_offset = 0;
+	d->posn.host_offset = 0;
+
+	k_spin_unlock(&d->lock, key);
```

Comment:
sorry, not sure I understand the use of locking well here. I looked through dma-trace.c and I couldn't understand what this lock is actually protecting. As for these `host_offset` fields - looks like in other locations they are accessed and modified without holding the lock?

Reply:
True, there is no need for the lock.

Reply:
yes, I will.